relay-server/src/actors/project_upstream.rs

```diff
@@ -44,10 +45,15 @@ pub struct GetProjectStates {
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct GetProjectStatesResponse {
+    /// Map of [`ProjectKey`] to [`ProjectState`] that was fetched from the upstream.
+    #[serde(default)]
+    pub(crate) configs: HashMap<ProjectKey, ErrorBoundary<Option<ProjectState>>>,
+    /// The [`ProjectKey`]'s that couldn't be immediately retrieved from the upstream.
     #[serde(default)]
-    pub configs: HashMap<ProjectKey, ErrorBoundary<Option<ProjectState>>>,
+    pub(crate) pending: Vec<ProjectKey>,
+    /// The global config fetched from the upstream.
     #[serde(default)]
-    pub pending: Vec<ProjectKey>,
+    pub(crate) global: Option<GlobalConfig>,
```

Comment:
Please, keep the naming consistent. Either `global` everywhere or `global_config`. 

```suggestion
    pub(crate) global_config: Option<GlobalConfig>,
```